sharedbuffers/mapped_struct.py

```diff
@@ -964,10 +966,10 @@ def _stable_hash(key):
             if expo < 0:
                 # A double's exponent is usually limited to [-1024, 1024]
                 expo += 0xFFFF
-            if math.isinf(mant):
-                mant = 1 if mant > 0 else -1
-            elif math.isnan(mant):
-                mant = 2
+            if isinf(mant):
+                mant = 1. if mant > 0 else -1.
+            elif isnan(mant):
```

Comment:
You have to import isinf/isnan from math in pure python or you'll break pure python. Also import frexp into a fast global to avoid the attribute access to the math module.

Reply:
`frexp` has a different calling convention than most other functions in that it takes an output parameter (a pointer to an int), so it'd look and work differently whether `cython.compiled` is true or not.

Furthermore, the standard doesn't really say what `frexp` has to return when the input float is +/- inf or NaN. The behaviour is unspecified. I think it's better if we just rely on Python's implementation handling all these subtleties.

Reply:
@llogiudice You can check tornado_bidder.py and look how random_random is imported.